### ISSUE TITLE
Projection in feet returned in metres

### ISSUE THREAD

**Issue by tr3vorm** [proj4js/proj4js#147]
If I perform a projection where the source is in metres, and the target is in 'us-ft' the result appears to be returned in metres. Is this expected behaviour and I need to convert back to feet manually, or is there something I am missing in my call?

``` javascript
proj4(proj4.defs['EPSG:3857'], proj4.defs('EPSG:3735')).forward([pnt.x, pnt.y]);
```

**Comment by calvinmetcalf** [comment]
let me rewrite this to make sure we are on the same page, do you get the same issue when you do

```js
var epsg3735 = '+proj=lcc +lat_1=40.03333333333333 +lat_2=38.73333333333333 +lat_0=38 +lon_0=-82.5 +x_0=600000 +y_0=0 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=us-ft +no_defs';
proj4('EPSG:3857', epsg3735).forward(pnt);
```

**Comment by calvinmetcalf** [comment]
can we get a test case, aka when pnt is equal to ? you expect ? but get ?

**Comment by kcarrier** [comment]
@calvinmetcalf - My name is Ken Carrier @tr3vorm has been helping me in trying to figure this out. Here is how we were testing and trying to determine a solution.

When using a product known as ArcMap I get the results below.

State Plane : 1475451.98515687, 398189.903394876
Decimal Degress : 39.0802596628748, -84.2369792985637
Degrees Minutes Seconds : 39°04'49" N; 84°14'13" W

When using @tr3vorm tool I get the results below.

NAD83 : 449718.3520304, 121369.3083012
WMAS : -9377218.06, 4733175.61
WGS84 : 39.0802666, -84.2369831

**Comment by kcarrier** [comment]
@calvinmetcalf - My name is Ken Carrier @tr3vorm has been working with me to troubleshoot this further. Here are the results.

ArcMap Results:
State Plane : 1475451.98515687, 398189.903394876
Decimal Degress : 39.0802596628748, -84.2369792985637
Degrees Minutes Seconds : 39°04'49" N; 84°14'13" W

@tr3vorm Results:
NAD83 : 449718.3520304, 121369.3083012
WGS84 : 39.0802666, -84.2369831

**Comment by calvinmetcalf** [comment]
tl;dr remove the reference to EPSG:3857 it's actually EPSG:4326

```js
proj4(proj4.defs['EPSG:3857'], proj4.defs('EPSG:3735')).forward([pnt.x, pnt.y]);
```

this converts from google spherical mercator meters to ohio state plane, you want to convert from wgs84 unprojected decrees (aka epsg:4326), since epst4326 is the default you can just do

```js
proj4(proj4.defs('EPSG:3735')).forward([pnt.x, pnt.y]);
```

which for `[-84.2369792985637, 39.0802596628748]` returns `[ 1475451.985156881, 398189.90339486714 ]`

**Comment by kcarrier** [comment]
@calvinmetcalf - Most of our maps are in WMAS or WKID:3857, I think he is getting the projection info from the map itself and passing that as a parameter/variable.

Here is his is how we pass the srid or WKID:
https://gist.github.com/kcarrier/77cdbfcd283be92ebd44

Here is @tr3vorm projection logic:
https://gist.github.com/kcarrier/522bcec861410c4e8580

**Comment by tr3vorm** [comment]
Thank you @calvinmetcalf - and yes, we are on the same page (re your first comment).

I get the base projection from the map (ArcGIS basemap), and this works fine for most of the target projections I have worked with (my only exception has been projections where grid shifts are used). 

Given the example provided by @kcarrier, is the result an expected one? I am researching the `EPSG:3857 it's actually EPSG:4326` statement, but **3857** is what my maps coordinates are projected in.

**Comment by calvinmetcalf** [comment]
WGS84 is a datum, a way of describing points on the sphere of earth, NAD83 is another one.  State plane (AKA EPSG:3735) and Spherical Web Mercator (aka EPSG: 3857) are projections for creating a 'projection' of the spherical earth onto a flat surface (your map).  

EPSG:4326 refers to WGS84 just with lat longs.

Spherical Web Mercator (aka EPSG:3857) is what the base map is drawn with, but the data you are projecting are unprojected lat longs, in other words despite the map being in drawn in 3857, the data you are dealing with is in EPSG:4326.  This is fairly common in web mapping where date is usually stored as unprojected lat longs and projected on the fly.

**Comment by kcarrier** [comment]
I cannot get your code to work correctly as you described above.

proj4(proj4.defs('EPSG:3735')).forward([-84.2369792985637, 39.0802596628748]);

returns

[449718.66451314633, 121368.52529180609]

Not sure what I could be doing wrong here

**Comment by calvinmetcalf** [comment]
Testing against http://cs2cs.mygeodata.eu that looks to be correct

**Comment by kcarrier** [comment]
Calvin,

I am confused because using the same coordinates earlier in this thread you said this:

proj4(proj4.defs('EPSG:3735')).forward([pnt.x, pnt.y]);
which for [-84.2369792985637, 39.0802596628748] returns [ 1475451.985156881, 398189.90339486714 ]

which is correct, it appear spatialreference.org returns the proper coordinate whereas epsg.io does not.

**Comment by kcarrier** [comment]
Is there a way to dynamically read the string from this url

http://spatialreference.org/ref/epsg/3735/proj4/

and have it return the string to use with your code which should be

'+proj=lcc +lat_1=40.03333333333333 +lat_2=38.73333333333333 +lat_0=38 +lon_0=-82.5 +x_0=600000 +y_0=0 +ellps=GRS80 +datum=NAD83 +to_meter=0.3048006096012192 +no_defs'

**Comment by calvinmetcalf** [comment]
http://cs2cs.mygeodata.eu/ is what I use when I need to check against a 3rd party to see what we should be getting, it returns `[449718.66451314633, 121368.52529180609]`

on the other hand 

```js
var epsg3735 = '+proj=lcc +lat_1=40.03333333333333 +lat_2=38.73333333333333 +lat_0=38 +lon_0=-82.5 +x_0=600000 +y_0=0 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=us-ft +no_defs';

proj4(epsg3735).forward( [-84.2369792985637, 39.0802596628748] )

```

return `[ 1475451.985156881, 398189.90339486714 ]` which is what arcmap returns so by a vote of 2 to 1 it sounds like `[ 1475451.985156881, 398189.90339486714 ]` is the correct answer, as for why your application is getting the wrong answer that depends how it's defining 'EPSG:3735'

**Comment by kcarrier** [comment]
Well I would like to try and make my application more dynamic, is there a way to pass a url to your proj4js dynamically and have it return the string from spatialreference.org so I do not have to hardcode the string, would make it more user friendly for others.

**Comment by kcarrier** [comment]
Using a geometry service proved a little difficult because the call is asynchronous and it would fire  but not in the proper order so sometimes depending on the length of time it would work and sometimes it would not, your method seems more stable.

**Comment by calvinmetcalf** [comment]
you can write your own function which downloads the the srs and then adds
it to the defs, proj4 intentionally does not do that due to the added
complexity and that you'd need to rely on that service going down.  we make
the assumption that in most applications you don't need all projections,
you just need your projection and looking that one up is easy.

**Comment by kcarrier** [comment]
That makes sense Calvin, is there an example of best practices for hardcoding this?

**Comment by calvinmetcalf** [comment]
so since epsg.org doesn't have cors enabled you'll want to do it with epsg.io and get the link via ``https://epsg.io/${code}.proj4`` and then just download via [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)
